src/components/RequesterPaysModal.tsx

```diff
@@ -0,0 +1,117 @@
+import { ButtonPrimary, ExternalLink, Modal, useUniqueId } from '@terra-ui-packages/components';
+import * as _ from 'lodash/fp';
+import React, { useState } from 'react';
+import { spinnerOverlay, VirtualizedSelect } from 'src/components/common';
+import { FormLabel } from 'src/libs/forms';
+import * as Nav from 'src/libs/nav';
+import { requesterPaysProjectStore } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { DEFAULT } from 'src/libs/utils';
+import { useWorkspaces } from 'src/workspaces/common/state/useWorkspaces';
+import { isGoogleWorkspace, WorkspaceWrapper } from 'src/workspaces/utils';
+
+const requesterPaysHelpInfo = (
+  <div style={{ paddingTop: '1rem' }}>
+    <ExternalLink href='https://support.terra.bio/hc/en-us/articles/360029801491'>
+      Why is a workspace required to access this data?
+    </ExternalLink>
+  </div>
+);
+
+interface RequesterPaysModalProps {
+  onDismiss: () => void;
+  onSuccess: (selectedGoogleProject: string) => void;
+}
+
+const RequesterPaysModal: React.FC<RequesterPaysModalProps> = ({ onDismiss, onSuccess }) => {
+  const { workspaces, loading } = useWorkspaces();
+  const billableWorkspaces = _.filter(
+    (workspace: WorkspaceWrapper) =>
+      isGoogleWorkspace(workspace) && (workspace.accessLevel === 'OWNER' || workspace.accessLevel === 'PROJECT_OWNER'),
+    workspaces
+  );
+  const selectId = useUniqueId('select');
```

Comment:
Replaced `IdContainer` with `useUniqueId`, which removes some of the deeper component nesting.